app/scripts/modules/core/delivery/executionGroup/ExecutionGroup.tsx

```diff
@@ -0,0 +1,239 @@
+import * as React from 'react';
+import * as ReactGA from 'react-ga';
+import { $rootScope, $timeout } from 'ngimport';
+import { IPromise } from 'angular';
+import { Subscription } from 'rxjs/Subscription';
+import { find, flatten, uniq } from 'lodash';
+
+import { AccountLabelColor } from 'core/account/AccountLabelColor';
+import { Application } from 'core/application/application.model';
+import { Execution } from './execution/Execution';
+import { IExecution } from 'core/domain';
+import { NextRunTag } from 'core/delivery/triggers/NextRunTag';
+import { Sticky } from 'core/utils/stickyHeader/Sticky';
+import { Tooltip } from 'core/presentation/Tooltip';
+import { TriggersTag } from 'core/delivery/triggers/TriggersTag';
+import { collapsibleSectionStateCache } from 'core/cache/collapsibleSectionStateCache';
+import { executionService } from 'core/delivery/service/execution.service';
+import { executionFilterModel } from 'core/delivery/filter/executionFilter.model';
+import { modalService } from 'core/modal.service';
+import { pipelineConfigService } from 'core/pipeline/config/services/pipelineConfig.service';
+import { stateService, stateParamsService } from 'core/state.service';
+
+import './executionGroup.less';
+
+interface IProps {
+  group: any;
+  application: Application;
+}
+
+interface IState {
+  deploymentAccounts: string[];
+  pipelineConfig: any;
+  triggeringExecution: boolean;
+  open: boolean;
+  poll: IPromise<any>;
+  canTriggerPipelineManually: boolean;
+  canConfigure: boolean;
+  showAccounts: boolean;
+}
+
+export class ExecutionGroup extends React.Component<IProps, IState> {
+  private strategyConfig: any;
+  private expandUpdatedSubscription: Subscription;
+
+  constructor(props: IProps) {
+    super(props);
+
+    this.strategyConfig = find(this.props.application.strategyConfigs.data, { name: this.props.group.heading });
+
+    const pipelineConfig = find(this.props.application.pipelineConfigs.data, { name: this.props.group.heading });
+    const sectionCacheKey = this.getSectionCacheKey();
+
+    this.state = {
+      deploymentAccounts: this.getDeploymentAccounts(),
+      triggeringExecution: false,
+      open: this.isShowingDetails() || !collapsibleSectionStateCache.isSet(sectionCacheKey) || collapsibleSectionStateCache.isExpanded(sectionCacheKey),
+      poll: null,
+      canTriggerPipelineManually: !!pipelineConfig,
+      canConfigure: !!(pipelineConfig || this.strategyConfig),
+      showAccounts: executionFilterModel.sortFilter.groupBy === 'name',
+      pipelineConfig,
+    };
+
+    $rootScope.$on('$stateChangeSuccess', () => {
```

Comment:
Yep, forgot to clean this up. Very good point to just move it to stateService and sub/unsub.